feat: add full storage database fault

phylisscity opened the pull request:
This PR adds the Valkey fill storage fault to simulate database storage exhaustion

Red-GV commented on sre/roles/faults/tasks/inject_valkey_fill_storage.yaml:
```diff
@@ -0,0 +1,34 @@
+    namespace: "{{ spec.workload.namespace }}"
+    patch:
+      - op: add
+        path: /spec/template/spec/containers/-
```
```suggestion
        path: /spec/template/spec/containers
```

phylisscity commented:
Updated per feedback:

- Changed incident number from 298 → 40
- Removed unused `storage_fill_percentage` parameter
- Added `storage_size_mb` as user-configurable input (default: 500MB)
- Updated validation schema in `argument_specs.yaml`
- Cleaned up spacing across all fault files
- Added smoke test for incident 40 in CI pipeline

The storage size is now configurable by users in the incident spec file while maintaining a default of 500MB.

Ready for review!

Red-GV commented on sre/roles/faults/tasks/inject_valkey_fill_storage.yaml:
Just add a single new line between each task for easier reading.

phylisscity commented on sre/roles/faults/tasks/inject_valkey_fill_storage.yaml:
```diff
@@ -0,0 +1,35 @@
+            - /bin/sh
+            - -c
+            - dd if=/dev/zero of=/tmp/filler bs=1M count={{ storage_size_mb }} && sleep infinity
+  when:
```
do you want me to remove the set storage size to fill task (lines 10-15) and remove line 33 (the dd command)?

Red-GV reviewed (approved):
LGTM
